src/plugins/intel_gpu/src/graph/layout_optimizer.cpp

```diff
@@ -1551,7 +1597,10 @@ format layout_optimizer::get_preferred_format(program_node& node) {
         } else if (use_onednn_impls && data_type_traits::is_i8_u8(layout.data_type) &&
             needs_all_usr_onednn_small_ic_to_blocked(node)) {
             // All user nodes are convolutions which satisfy options for onednn first conv
-            expected = format::bs_fs_yx_bsv8_fsv4;
+            if (layout.format.spatial_num() == 2)
+                expected = format::bs_fs_yx_bsv8_fsv4;
+            else if (layout.format.spatial_num() == 3)
+                expected = format::bs_fs_zyx_bsv8_fsv4;
```

Comment:
What about defining a helper function such as below? I think this helper function will make the code much readable..
`expected = get_layout(format::bs_fs_yx_bsv8_fsv4, layout.format.spatial_num())`